src/games/strategy/triplea/ai/proAI/ProAI.java

```diff
@@ -346,14 +354,19 @@ public Territory retreatQuery(final GUID battleID, final boolean submerge, final
       return null;
     }
 
-    // If I'm attacker, have more unit strength, and isn't land battle with only air left then don't retreat
+    // If attacker with more unit strength or strafing and isn't land battle with only air left then don't retreat
     final boolean isAttacker = player.equals(battle.getAttacker());
     final List<Unit> attackers = (List<Unit>) battle.getAttackingUnits();
     final List<Unit> defenders = (List<Unit>) battle.getDefendingUnits();
     final double strengthDifference = battleUtils.estimateStrengthDifference(battleTerritory, attackers, defenders);
+    boolean isStrafing = false;
+    if (isAttacker && storedStrafingTerritories.contains(battleTerritory)) {
```

Comment:
Can simplify this:
`boolean isStrafing = (isAttacker && stored..contains(battleTerritory));`